styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -191,7 +232,28 @@ private void updateStats(Map<String, Resource> resources,
 
   private boolean limitAndDequeue(Map<String, Resource> resources,
       Map<WorkflowId, Set<String>> workflowResourceReferences,
-      Map<String, Long> currentResourceUsage, InstanceState instance) {
+      Map<String, Long> currentResourceUsage, InstanceState instance,
+      CompletionStage<Optional<ExecutionBlocker>> executionBlockerFuture) {
+
+    // Check for execution blockers
+    Optional<ExecutionBlocker> blocker = Optional.empty();
+    try {
+      blocker = executionBlockerFuture.toCompletableFuture().get(30, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      LOG.debug("Interrupted");
+      return false;
+    } catch (ExecutionException | TimeoutException e) {
+      LOG.warn("Failed to check execution blockers for {}", instance.workflowInstance(), e);
```

Comment:
`s/blockers/blocker`, also I think the log can be clearer that we will continue as if there were no blocker.

Reply:
👍